### DISCUSSION TITLE
Form with dynamic table checkbox selection

### DISCUSSION THREAD

**Discussion by dyljhd** [react-hook-form/react-hook-form#12075] [Q&A] [answered]
I want to know if the following is possible with React Hook Form.

Here is a screenshot of the form I have:
![image](https://github.com/react-hook-form/react-hook-form/assets/143370090/661eb7c9-be9b-4528-adf4-a3572280d12f)

I have a document type and a document name field, which are both just strings - nice and simple.

But, I have a dynamic table - by dynamic I mean the ability to change (order) by the way of column sorting, pagination, or potentially filtering by a value for a column etc. 
Regardless of what the filter is, it causes the table to reorder rows or show/hide rows (when paginating).

Each of these rows contains a numerical ID attribute (something to uniquely identify it) as well as attributes for the values seen in the table.

Can I get an output of a list of numerical IDs (e.g. `[11,24,36]`) that have been selected via the checkboxes from this table?

I could have missed something, but everything I have seen seems to work with array indexes (e.g. `useFieldArray`), which doesn't seem like it would work here.

...and if not possible with vanilla React Hook Form, is there a way to integrate a custom hook solution (that I have already built) that would store the list of numerical IDs in React state that could then sync with React Hook Form's form state to work with resetting of the form, validation and submission etc.

Any further questions, just ask. Any help with this would be much appreciated. Thanks in advance! :)

**Comment by dyljhd** [comment, marked as answer]
Nevermind, I came back to the problem the next day with a fresh mind and solved it fairly easily - silly me :').

**Reply by dyljhd** [reply]
The solution is quite a lot to explain, but if anyone comes across this problem, just reply here, and I can assist :)

**Reply by vietvh-3042** [reply]
Please help me validate with table pagination :(
